runhouse/servers/obj_store.py

```diff
@@ -572,11 +580,40 @@ def delete_for_env_servlet_name(env_servlet_name: str, key: Any):
     def delete_local(self, key: Any):
         self.pop_local(key)
 
+    def delete_env(self, key: Any):
+        from runhouse.globals import env_servlets
+
+        # clear keys in the env servlet
+        env_servlet_keys = list(self._kv_store.keys()) if self.has_local_storage else []
+        self.clear_for_env_servlet_name(key)
+
+        # delete the env servlet actor
+        if key in env_servlets:
+            actor = env_servlets[key]
+            ray.kill(actor)
+
+            del env_servlets[key]
+
+        # delete the local key
+        if self.contains_local(key):
+            self.delete_local(key)
+
+        self.remove_env_servlet_name(key)
+        return env_servlet_keys
+
     def delete(self, key: Union[Any, List[Any]]):
         keys_to_delete = [key] if isinstance(key, str) else key
+        deleted_keys = []
+
         for key_to_delete in keys_to_delete:
-            if self.contains_local(key_to_delete):
+            if key_to_delete in deleted_keys:
```

Comment:
ah i understand, technically you could have already deleted something they also wanted to delete

Reply:
nice